src/events/actions.py

```diff
@@ -39,15 +39,22 @@ def _on_get_jdbc_endpoint(self, event: ActionEvent):
         if not self.workload.ready():
             event.fail("The action failed because the workload is not ready yet.")
             return
-        if (
-            not self.get_app_status(
-                s3_info=self.context.s3, service_account=self.context.service_account
-            )
-            != Status.ACTIVE
-        ):
+        if self.get_app_status() != Status.ACTIVE.value:
             event.fail("The action failed because the charm is not in active state.")
             return
-        result = {"endpoint": self.workload.get_jdbc_endpoint()}
+
+        if self.context.is_ha_enabled():
+            address = self.context.zookeeper.uris
+            # FIXME: Get this value from self.context.zookeeper.uris when znode created by
+            # zookeeper charm has enough permissions for Kyuubi to work
+            namespace = HA_ZNODE_NAME
+            if not address.endswith("/"):
+                address += "/"
+            endpoint = f"jdbc:hive2://{address};serviceDiscoveryMode=zooKeeper;zooKeeperNamespace={namespace}"
```

Comment:
**suggestion** I would put this business logic more in the `kyuubi` manager, rather than in the `handlers`. 

We could probably just expose a method in the manager that is `get_endpoint`, and there place the business logic, also based on the HA/non-HA logic